lib/modules/manager/devbox/artifacts.ts

```diff
@@ -1,42 +1,64 @@
 import is from '@sindresorhus/is';
+import { SemVer } from 'semver';
 import { logger } from '../../../logger';
 import { exec } from '../../../util/exec';
+import { resolveConstraint } from '../../../util/exec/containerbase';
 import type { ExecOptions } from '../../../util/exec/types';
 import { getSiblingFileName, readLocalFile } from '../../../util/fs';
 import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
 
-export async function updateArtifacts(
-  updateConfig: UpdateArtifact,
-): Promise<UpdateArtifactsResult[] | null> {
-  const lockFileName = getSiblingFileName(
-    updateConfig.packageFileName,
-    'devbox.lock',
-  );
+export async function updateArtifacts({
+  config: { constraints, env, isLockFileMaintenance, updateType },
+  packageFileName,
+  updatedDeps,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  const lockFileName = getSiblingFileName(packageFileName, 'devbox.lock');
   const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
   if (!existingLockFileContent) {
     logger.debug('No devbox.lock found');
     return null;
   }
+
+  let supportsNoInstall = true;
+  if (constraints?.devbox) {
+    const constraintVersion = await resolveConstraint({
+      toolName: 'devbox',
+      constraint: constraints?.devbox,
+    });
+    supportsNoInstall = new SemVer(constraintVersion).compare('0.14.0') >= 0;
```

Comment:
use use something like `semver.satisfies(version, '>=0.14.0')` so no need to resolve the real version used .